### PR TITLE
add rttr/0.9.6

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **rttr/0.9.6**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/rttr/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_rtti": False,
+    }
+
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows" or self.options.shared:
```
Comment (danimtb) [inline recipes/rttr/all/conanfile.py]: fPIC flag does not apply when building in static, so there is no need to add this check
```suggestion
        if self.settings.os == "Windows":
```

File: recipes/rttr/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        cmake.definitions["BUILD_UNIT_TESTS"] = False
+        cmake.definitions["BUILD_WITH_RTTI"] = self.options.with_rtti
+        cmake.definitions["BUILD_PACKAGE"] = False
+
+        if self.options.shared:
+            cmake.definitions["BUILD_RTTR_DYNAMIC"] = True
+            cmake.definitions["BUILD_STATIC"] = False
+        else:
+            cmake.definitions["BUILD_STATIC"] = True
+            cmake.definitions["BUILD_RTTR_DYNAMIC"] = False
```
Comment (danimtb) [inline recipes/rttr/all/conanfile.py]: You can simplify this logic with:
```
cmake.definitions["BUILD_RTTR_DYNAMIC"] = self.options.shared
cmake.definitions["BUILD_STATIC"] = not self.options.shared
```

**Comment by intelligide** [comment]
3rd party libs are not used in build. Catch is the test framework, nonius the benchmarking framework, and rapidjson is used in only one example. (and the recipe doesn't build tests, benchmarks and examples)

**Comment by danimtb** [comment]
Thanks for the clarification. I thought that rapidjson was needed at least, but it is not 👍

**Pull request merged**